gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequest.java

```diff
@@ -38,280 +30,163 @@
-   * ChangeRequest}.
-   */
-  public List<RecordSet> deletions() {
-    return deletions;
+  static Function<Change, ChangeRequest> fromPbFunction(final Dns dns, final String zoneName) {
+    return new Function<Change, ChangeRequest>() {
+      @Override
+      public ChangeRequest apply(com.google.api.services.dns.model.Change pb) {
+        return ChangeRequest.fromPb(dns, zoneName, pb);
+      }
+    };
   }
 
   /**
-   * Returns the id assigned to this {@code ChangeRequest} by the server.
+   * Returns the name of the {@link Zone} associated with this change request.
    */
-  public String id() {
-    return id;
+  public String zoneName() {
+    return this.zoneName;
   }
 
   /**
-   * Returns the time when this {@code ChangeRequest} was started by the server.
+   * Returns the {@link Dns} service object associated with this change request.
    */
-  public Long startTimeMillis() {
-    return startTimeMillis;
+  public Dns dns() {
+    return dns;
   }
 
   /**
-   * Returns the status of this {@code ChangeRequest}.
+   * Applies this change request to a zone that it is associated with.
    */
-  public Status status() {
-    return status;
-  }
-
-  com.google.api.services.dns.model.Change toPb() {
-    com.google.api.services.dns.model.Change pb =
-        new com.google.api.services.dns.model.Change();
-    // set id
-    if (id() != null) {
-      pb.setId(id());
-    }
-    // set timestamp
-    if (startTimeMillis() != null) {
-      pb.setStartTime(ISODateTimeFormat.dateTime().withZoneUTC().print(startTimeMillis()));
-    }
-    // set status
-    if (status() != null) {
-      pb.setStatus(status().name().toLowerCase());
-    }
-    // set a list of additions
-    pb.setAdditions(Lists.transform(additions(), RecordSet.TO_PB_FUNCTION));
-    // set a list of deletions
-    pb.setDeletions(Lists.transform(deletions(), RecordSet.TO_PB_FUNCTION));
-    return pb;
+  public ChangeRequest applyTo(Dns.ChangeRequestOption... options) {
+    return dns.applyChangeRequest(zoneName, this, options);
   }
 
-  static ChangeRequest fromPb(com.google.api.services.dns.model.Change pb) {
-    Builder builder = builder();
-    if (pb.getId() != null) {
-      builder.id(pb.getId());
-    }
-    if (pb.getStartTime() != null) {
-      builder.startTimeMillis(DateTime.parse(pb.getStartTime()).getMillis());
-    }
-    if (pb.getStatus() != null) {
-      // we are assuming that status indicated in pb is a lower case version of the enum name
-      builder.status(ChangeRequest.Status.valueOf(pb.getStatus().toUpperCase()));
-    }
-    if (pb.getDeletions() != null) {
-      builder.deletions(Lists.transform(pb.getDeletions(), RecordSet.FROM_PB_FUNCTION));
-    }
-    if (pb.getAdditions() != null) {
-      builder.additions(Lists.transform(pb.getAdditions(), RecordSet.FROM_PB_FUNCTION));
-    }
-    return builder.build();
+  @Override
+  public Builder toBuilder() {
```

Comment:
Given the amount of casting you could consider using a `ChangeRequest other = (ChangeRequest) obj` variable.